maths/binary_exponent.py

```diff
@@ -0,0 +1,31 @@
+def bin_exp(num: int, expo: int) -> int:
+    """
+    Calculate the exponent of 'num' over 'expo' and return a positive number.
+    This function does binary exponential hence O(log N) time complexity.
+    So, It is faster than normal exponential function for huge numbers.
+    >>> bin_exp(2, 5)
+    32
+    >>> bin_exp(1, 0)
+    1
+    >>> bin_exp(0, -2)
+    'expo must not be negative'
+    >>> bin_exp(-2, -2)
+    'expo must not be negative'
+    """
+    result = 1
+    if expo < 0:
+        return 'expo must not be negative'
```

Comment:
Line 1 promises that this function will return an int so we can not return a str.  Please raise the ValueError instead.  Search this repo for ValueError to see other algorithms with doctests that deal with exceptions.

Reply:
Oh okay sir, I'll do it.

Reply:
Nice!  Mypy even agrees with me https://travis-ci.com/github/TheAlgorithms/Python/builds/182944036#L459